tests/keras/backend/reference_operations.py

```diff
@@ -166,3 +166,53 @@ def rnn(x, w, init, go_backwards=False, mask=None, unroll=False, input_length=No
         h.append(h_t + h_t1)
 
     return o[-1], np.stack(o, axis=1), np.stack(h, axis=1)
+
+
+def basics(function_name, x, **kwargs):
```

Comment:
Wouldn't it be preferable to have separate functions for these, instead of a single function with multiple if/else statements?